spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/support/UnexpectedTaskExecutionException.java

```diff
@@ -0,0 +1,199 @@
+
+	/**
+	 * The recorded exit code for the task.
+	 */
+	private Integer exitCode = -1;
+
+	/**
+	 * User defined name for the task.
+	 */
+	private String taskName;
+
+	/**
+	 * Time of when the task was started.
+	 */
+	private Date startTime;
+
+	/**
+	 * Timestamp of when the task was completed/terminated.
+	 */
+	private Date endTime;
+
+	/**
+	 * Message returned from the task or stacktrace.
+	 */
+	private String exitMessage;
+
+	/**
+	 * Id assigned to the task by the platform.
+	 */
+	private String externalExecutionId;
+
+	/**
+	 * Error information available upon the failure of a task.
+	 */
+	private String errorMessage;
+
+	/**
+	 * The arguments that were used for this task execution.
+	 */
+	private ArrayList<String> arguments;
+
+	/**
+	 * Constructs an UnexpectedTaskExecutionException with the specified
+	 * detail message.
+	 *
+	 * @param message the detail message
+	 */
+	public UnexpectedTaskExecutionException(String message) {
+		super(message);
+	}
+
+	/**
+	 * Constructs an UnexpectedTaskExecutionException with the specified
+	 * detail message, cause and exitCode.
+	 *
+	 * @param message the detail message
+	 * @param cause   the cause which leads to this exception
+	 */
+	public UnexpectedTaskExecutionException(String message, Throwable cause) {
+		super(message, cause);
+	}
+
+	/**
+	 * Constructs an UnexpectedTaskExecutionException with the specified
+	 * detail message and taskExecution.
+	 *
+	 * @param message       the detail message
+	 * @param taskExecution the taskExecution of the task
+	 */
+	public UnexpectedTaskExecutionException(String message, TaskExecution taskExecution) {
+		this(message);
+		assignTaskExecutionFields(taskExecution);
+	}
+
+	/**
+	 * Constructs an UnexpectedTaskExecutionException with the specified
+	 * detail message, cause and taskExecution.
+	 *
+	 * @param message       the detail message
+	 * @param cause         the cause which leads to this exception
+	 * @param taskExecution the taskExecution of the task
+	 */
+	public UnexpectedTaskExecutionException(String message, Throwable cause, TaskExecution taskExecution) {
+		this(message, cause);
+		assignTaskExecutionFields(taskExecution);
+	}
+
+	/**
+	 * Assigns the task execution fields to this exception.
+	 *
+	 * @param taskExecution the task execution of which the fields should be assigned to this exception
+	 */
+	private void assignTaskExecutionFields(TaskExecution taskExecution) {
+		if(taskExecution != null) {
+			executionId = taskExecution.getExecutionId();
+			parentExecutionId = taskExecution.getParentExecutionId();
+			exitCode = taskExecution.getExitCode();
+			taskName = taskExecution.getTaskName();
+			startTime = taskExecution.getStartTime();
+			endTime = taskExecution.getEndTime();
+			externalExecutionId = taskExecution.getExternalExecutionId();
+			errorMessage = taskExecution.getErrorMessage();
+			exitMessage = taskExecution.getExitMessage();
+			arguments = new ArrayList<>(taskExecution.getArguments());
```

Comment:
Refresh my memory, are these arguments already obfuscated at this point?

Reply:
I actually don't know if they are sanitized. It depends on what is returned by `this.taskExplorer.getTaskExecution(this.executionId)`